packages/web-components/src/utils/date-utils.ts

```diff
@@ -171,15 +175,22 @@ export function validate(
   // We don't know the upper limit on days until we know the month
   if (!monthYearOnly && (day < minMonths || day > daysForSelectedMonth)) {
     component.invalidDay = true;
-    component.error = `Please enter a day between ${minMonths} and ${daysForSelectedMonth}`;
+    component.error = dayError;
+
+    // We need a valid month in order to determine the day
+    if (!month || month < minMonths || month > maxMonths) {
+      component.invalidMonth = true;
+      component.error = monthError;
+    }
```

Comment:
Having this extra check inside the day check is the main change here. The other changes are refactoring to use shared consts.